src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Acquisition/MissingSetupComponentRegistrationService.cs

```diff
@@ -38,6 +50,9 @@ internal class MissingSetupComponentRegistrationService : IMissingSetupComponent
         private IVsSolution? _vsSolution;
         private bool? _isVSFromPreviewChannel;
 
+        private readonly object _lock = new();
```

Comment:
This needs to be static in order to protect `_netCoreRegistryKeyValues` properly, as that field is static.